api/app/src/routes/user.ts

```diff
@@ -112,45 +114,116 @@ router.get(
   })
 );
 
+async function revokeUserProviderAccess(
+  connectedUser: ConnectedUser,
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  provider: any
+): Promise<void> {
+  const providerOAuth2 = providerOAuth2OptionsSchema.safeParse(provider);
+  // TODO #249: implement garmin revoke support
+  // const providerOAuth1 = providerOAuth1OptionsSchema.safeParse(
+  //   req.query.provider
+  // );
+  if (providerOAuth2.success) {
+    await Constants.PROVIDER_OAUTH2_MAP[providerOAuth2.data].revokeProviderAccess(connectedUser);
+  } else if (provider === PROVIDER_APPLE) {
+    const apple = new Apple();
+    await apple.revokeProviderAccess(connectedUser);
+    // } else if (providerOAuth1.success) {
+    //   // await Constants.PROVIDER_OAUTH1_MAP[
+    //   //   providerOAuth1.data
+    //   // ].deregister(connectedUser);
+  } else {
+    throw new BadRequestError(`Provider not supported: ${provider}`);
+  }
+}
+
+async function revokeToken(req: Request, res: Response) {
+  const userId = getUserIdFrom("query", req).orFail();
+  const cxId = getCxIdOrFail(req);
+  const connectedUser = await getConnectedUserOrFail({ id: userId, cxId });
+  await revokeUserProviderAccess(connectedUser, req.query.provider);
+  return res
+    .status(status.OK)
+    .json({ message: `Access token for ${req.query.provider} has been revoked.` });
+}
+
 /** ---------------------------------------------------------------------------------------
  * DELETE /user/revoke
+*
+* Revoke access to a provider
+*
+* @param   {string}  req.query.provider    The provider to revoke access.
+* @param   {string}  req.query.userId      The internal user ID.
+
+* @return  {{success: boolean}}      If successfully removed.
+*/
+router.delete(
+  "/:id/revoke",
+  asyncHandler(async (req: Request, res: Response) => {
+    return revokeToken(req, res);
+  })
+);
+
+/**
+ * @deprecated use /:id/revoke instead.
+ *
+ * DELETE /user/revoke
  *
  * Revoke access to a provider
  *
  * @param   {string}  req.query.provider    The provider to revoke access.
  * @param   {string}  req.query.userId      The internal user ID.
-
+ *
  * @return  {{success: boolean}}      If successfully removed.
  */
 router.delete(
   "/revoke",
+  asyncHandler(async (req: Request, res: Response) => {
+    return revokeToken(req, res);
```

Comment:
I missed this on our huddle... this also needs update, since the user id is defined differently on `/revoke` and `/:id/revoke`

Maybe get those on each route and pass as a parameter to the `revokeToken` function?